autoagora_agents/algorithm.py

```diff
@@ -0,0 +1,432 @@
+# Copyright 2022-, Semiotic AI, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+import torch
+from torch import optim
+
+import experiment
+from autoagora_agents import buffer
+from autoagora_agents.distribution import distributionfactory
+
+
+class Algorithm(ABC):
+    def __init__(self, *, group: str, i: int) -> None:
+        self.niterations = 0
+        self.nupdates = 0
+        self.group = group
+        self.i = i
+        self.name = f"{group}_{i}"
+
+    def reset(self) -> None:
+        self.niterations = 0
+
+    def update(self):
+        self.nupdates += 1
+
+    @abstractmethod
+    def __call__(
+        self,
+        *,
+        observation: np.ndarray,
+        action: np.ndarray,
+        reward: float,
+        done: bool,
+    ) -> np.ndarray:
+        pass
```

Comment:
this is a base class, so some nice class description and description of (even abstract) methods would be useful ;)

Reply:
Good catch! Forgot to document this. Will take care of that.

Reply:
46549629cb729eb7325e9552be6521fab8ef711c